frontend/src/app/containers/App.js

```diff
@@ -118,36 +119,52 @@ function subscribeToBasket(email, callback) {
   });
 }
 
-export default connect(
-  state => ({
-    addon: state.addon,
-    experiments: experimentSelector(state),
-    experimentsLoaded: isExperimentsLoaded(state.experiments),
-    getExperimentBySlug: slug =>
-      getExperimentBySlug(state.experiments, slug),
-    hasAddon: state.addon.hasAddon,
-    installed: getInstalled(state.addon),
-    installedAddons: state.addon.installedAddons,
-    installedLoaded: isInstalledLoaded(state.addon),
-    isDev: state.browser.isDev,
-    isExperimentEnabled: experiment =>
-      isExperimentEnabled(state.addon, experiment),
-    isAfterCompletedDate,
-    isFirefox: state.browser.isFirefox,
-    isMinFirefox: state.browser.isMinFirefox,
-    routing: state.routing
-  }),
-  dispatch => ({
-    navigateTo: path => dispatch(routerPush(path)),
-    enableExperiment: experiment => enableExperiment(dispatch, experiment),
-    disableExperiment: experiment => disableExperiment(dispatch, experiment),
-    requireRestart: () => dispatch(addonActions.requireRestart()),
-    setHasAddon: installed => {
-      dispatch(addonActions.setHasAddon(installed));
-      if (!installed) { pollAddon(); }
-    }
-  }),
-  (stateProps, dispatchProps, ownProps) => Object.assign({
```

Comment:
This was getting unwieldy. I broke into parts to make it clearer and to make the actual `connect` call a one-liner.